hastory-data-gen/src/Data/Hastory/Gen.hs

```diff
@@ -30,9 +30,7 @@ instance GenValid UserForm where
   shrinkValid = shrinkValidStructurallyWithoutExtraFiltering
 
 instance GenValid Username where
-  genValid =
-    mkUsername <$>
-    (genValid `suchThat` \rawUsername -> notNull rawUsername && T.all isAlphaNum rawUsername)
+  genValid = Username <$> (genValid `suchThat` \text -> notNull text && T.all isAlphaNum text)
```

Comment:
This would be much much faster: https://hackage.haskell.org/package/genvalidity-text-0.7.0.2/docs/Data-GenValidity-Text.html#v:genTextBy